var/spack/repos/builtin/packages/nix/package.py

```diff
@@ -109,7 +110,7 @@ def installcheck(self):
 
     @run_after("install")
     def installcheck_clean(self):
-        if self.test_path:
+        if hasattr(self, "test_path"):
```

Comment:
Should this not also check what the attribute contains?

Reply:
What should I check for? I don't see how it could ever get a value except a directory path from the previous method.